### PR TITLE
Added BillingSpecifiedPeriod

### PR DESCRIPTION
**Pull request by pkurz**
Added field BG-14

### CODE REVIEW

**Comment by stephan-nowecor** [comment]
How would you handle if these parameters are set when writing ZUGFeRD 1.x files? Just ignore it or throw an exception?

**Comment by stephanstapel** [comment]
Two questions:
How would you handle if these parameters are set when writing ZUGFeRD 1.x files? Just ignore it or throw an exception?
Does the specification allow to only set start date or end date? Currently the code checks for validity of both fields

**Comment by pkurz** [comment]
1. the fields are indeed in 1.x available, but there is start and end date necessary if you specify the BillingSpecifiedPeriod
2. just rechecked the 2.1.1 specification and i misread. The start and end dates are 0..1, so can be left out.

**Comment by stephanstapel** [comment]
ok, thanks.
Would you mind adding the reader for version 2.1 and writer and writer for version 2.0 and 1.x?

**Comment by pkurz** [comment]
Yes i can, reader only for version 2.1?

**Comment by stephanstapel** [comment]
There was a typo in my answer :)
reader and writer for 1.x and 2.0 would be great. I guess it'd be just copy and paste?!

**Comment by pkurz** [comment]
Yes should be copy and paste, will update in the course of the day

**Comment by pkurz** [comment]
@stephanstapel have a look, should be fine now 👍

File: ZUGFeRD/InvoiceDescriptor21Writer.cs [hunk 1]
```diff
@@ -769,21 +780,21 @@ private void _writeOptionalTaxes(ProfileAwareXmlTextWriter writer)
...
-                writer.WriteAttributeString("currencyID", this.Descriptor.Currency.EnumToString());
+                //writer.WriteAttributeString("currencyID", this.Descriptor.Currency.EnumToString());
                 writer.WriteValue(_formatDecimal(tax.BasisAmount));
                 writer.WriteEndElement(); // !BasisAmount
 
                 if (tax.AllowanceChargeBasisAmount != 0)
                 {
                     writer.WriteStartElement("ram:AllowanceChargeBasisAmount");
-                    writer.WriteAttributeString("currencyID", this.Descriptor.Currency.EnumToString());
+                    //writer.WriteAttributeString("currencyID", this.Descriptor.Currency.EnumToString());
```
Comment (stephanstapel) [inline ZUGFeRD/InvoiceDescriptor21Writer.cs:797]: is it really correct to comment out the currencyID Completely?

Comment (pkurz) [reply ZUGFeRD/InvoiceDescriptor21Writer.cs:797]: whoops, good catch, thats my local change for the validators. Going to revert that real fast.

**Pull request merged**